beetsplug/parentwork.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+# This file is part of beets.
+# Copyright 2017, Dorian Soergel.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Gets work title, disambiguation, parent work and its disambiguation,
+composer, composer sort name and performers
+"""
+
+from __future__ import division, absolute_import, print_function
+
+from beets import ui
+from beets.plugins import BeetsPlugin
+from beets.dbcore import types
+
+import musicbrainzngs
+
+
+class ParentWorkPlugin(BeetsPlugin):
+
+    def __init__(self):
+        super(ParentWorkPlugin, self).__init__()
+        self.import_stages = [self.imported]
+
+    def commands(self):
+        cmd = ui.Subcommand('parentwork',
+                            help=u'fetches parent works, composers \
+                                and performers')
+        cmd.func = self.command
+        return [cmd]
+
+    def command(self, lib, opts, args):
+        self.find_work(lib.items(ui.decargs(args)))
+
+    item_types = {
+        'parent_work':          types.STRING,
+        'parent_work_disambig': types.STRING,
+        'work':                 types.STRING,
+        'work_disambig':        types.STRING,
+        'performer':            types.STRING,
+        'performer_sort':       types.STRING,
+        'parent_composer':      types.STRING,
+        'parent_composer_sort': types.STRING}
+
+    def imported(self, session, task):
+        self.find_work(task.items)
+
+    def find_work(self, items):
+    
+        def find_parentwork(work_id):
```

Comment:
Here is the error if I put the function find_parentwork inside the class but outside (and just before) the find_work function. I just copy-pasted it and arranged the indentation.  
Traceback (most recent call last):
  File "/usr/bin/beet", line 11, in <module>
    load_entry_point('beets==1.4.4', 'console_scripts', 'beet')()
  File "/usr/lib/python3.6/site-packages/beets/ui/__init__.py", line 1228, in main
    _raw_main(args)
  File "/usr/lib/python3.6/site-packages/beets/ui/__init__.py", line 1215, in _raw_main
    subcommand.func(lib, suboptions, subargs)
  File "/usr/lib/python3.6/site-packages/beetsplug/parentwork.py", line 43, in command
    self.find_work(lib.items(ui.decargs(args)))
  File "/usr/lib/python3.6/site-packages/beetsplug/parentwork.py", line 128, in find_work
    work_info = find_parentwork(work_id)
NameError: name 'find_parentwork' is not defined